src/hooks/useComponentVisible.ts

```diff
@@ -0,0 +1,23 @@
+import { useState, useEffect, useRef } from 'react';
+
+export default function useComponentVisible(initialIsVisible: any) {
+  const [isComponentVisible, setIsComponentVisible] = useState(initialIsVisible);
+  const ref = useRef(null);
+
+  const handleClickOutside = (event: any) => {
```

Comment:
Google event type

Reply:
removed. This file have already deleted )